### PR TITLE
Apps

### PR DESCRIPTION
**Pull request by sergisanrama**

### CODE REVIEW

File: staff/sergi-sanrama/spotify-app/scripts/main.js [hunk 1]
```diff
@@ -0,0 +1,92 @@
+// my presentation logic
```
Comment (ghost) [inline staff/sergi-sanrama/spotify-app/scripts/main.js]: use strict

File: staff/sergi-sanrama/spotify-app/scripts/main.js [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+// optional, reduce the size of the restaurants loaded in memory
+// restaurants.splice(100);
+
+var search = new SearchPanel();
+
+search.onSearch(function (query) {
+
+    logic.searchArtists(query)
+        .then(function (artists) {
+            resultsArtists.updateResults(artists.map(function (artist) {
```
Comment (ghost) [inline staff/sergi-sanrama/spotify-app/scripts/main.js]: You should also include the artist's image (you could pass the array unchanged and the manage it later).

File: staff/sergi-sanrama/spotify-app/scripts/main.js [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+                return {
+                    id: artist.id,
+                    text: artist.name,
+                };
+            }));
+            detailContainer.clear();
+        })
+        .catch(function (error) {
+            alert('Sorry, we have temporary problem, try again later.');
```
Comment (ghost) [inline staff/sergi-sanrama/spotify-app/scripts/main.js]: Maybe a more detailed message could be useful.

File: staff/sergi-sanrama/spotify-app/scripts/main.js [hunk 4]
```diff
@@ -0,0 +1,92 @@
...
+
+var resultsArtists = new ResultsList();
+
+var DEFAULT_IMAGE = 'https://i.pinimg.com/originals/37/2a/2d/372a2d5e8a32991bb19982271d0762fe.jpg';
+
+resultsArtists.onItemClick(function (id) {
+
+    logic.retrieveAlbumsByArtistId(id)
+        .then(function (albums) {
+            resultsAlbums.updateResults(albums.map(function (album) {
```
Comment (ghost) [inline staff/sergi-sanrama/spotify-app/scripts/main.js]: you should pass the entire array and manage it later (for example to retrieve the image).

File: staff/sergi-sanrama/spotify-app/scripts/main.js [hunk 5]
```diff
@@ -0,0 +1,92 @@
...
+                return {
+                    id: album.id,
+                    text: album.name,
+                };
+            }));
+            detailContainer.clear();
+        })
+        .catch(function (error) {
+            alert('Sorry, we have temporary problem, try again later.');
```
Comment (ghost) [inline staff/sergi-sanrama/spotify-app/scripts/main.js]: A more detailed message could be useful.

File: staff/sergi-sanrama/spotify-app/scripts/main.js [hunk 6]
```diff
@@ -0,0 +1,92 @@
...
+        });
+});
+var resultsAlbums = new ResultsList();
```
Comment (ghost) [inline staff/sergi-sanrama/spotify-app/scripts/main.js]: You could overwrite the updateResults function in order to show more information (images and other text).

**Review by ghost — CHANGES_REQUESTED** [review]
Some changes.

File: staff/sergi-sanrama/spotify-app/scripts/logic.js [hunk 7]
```diff
@@ -1,5 +1,5 @@
 var logic = {
```
Comment (gbaste) [inline staff/sergi-sanrama/spotify-app/scripts/logic.js]: Hi Sergi, you need add 'use strict' at the beginning of the js page

File: staff/sergi-sanrama/arrays/scripts/most-frequent.js [hunk 8]
```diff
@@ -0,0 +1,20 @@
...
+//console.log(most.length === 1);
+//console.log(most[0] === 2);
+
+function mostFrequent(array) {
+    // returns and array with the most frequent elements
+    var numberArray = [];
+    for(var i = 0; i < array.length; i++){
+        for(var j = i + 1; j < array.length; j++){
+            if(array[i] === array[j]){
+                numberArray.push(array[i])
```
Comment (gbaste) [inline staff/sergi-sanrama/arrays/scripts/most-frequent.js:16]: Need add semicolons ;;;;;

File: staff/sergi-sanrama/spotify-app/scripts/logic.js (hunk 7 above)
Comment (sergisanrama) [reply staff/sergi-sanrama/spotify-app/scripts/logic.js]: fixed

File: staff/sergi-sanrama/spotify-app/scripts/main.js (hunk 1 above)
Comment (sergisanrama) [reply staff/sergi-sanrama/spotify-app/scripts/main.js]: fixed

File: staff/sergi-sanrama/arrays/scripts/most-frequent.js (hunk 8 above)
Comment (sergisanrama) [reply staff/sergi-sanrama/arrays/scripts/most-frequent.js:16]: fixed

File: staff/sergi-sanrama/spotify-app/scripts/main.js (hunk 2 above)
Comment (sergisanrama) [reply staff/sergi-sanrama/spotify-app/scripts/main.js]: implemented

File: staff/sergi-sanrama/spotify-app/scripts/main.js (hunk 4 above)
Comment (sergisanrama) [reply staff/sergi-sanrama/spotify-app/scripts/main.js]: ready

**Pull request merged**